astropy/io/fits/fitsrec.py

```diff
@@ -614,7 +618,10 @@ def __del__(self):
             if self.dtype.fields is not None:
                 for col in self._col_weakrefs:
                     if col.array is not None:
-                        col.array = col.array.copy()
+                        # We use [:] but not copy() here so that the new
+                        # col.array is a view of the same region in memory
+                        # but a differe Python object.
+                        col.array = col.array[:]
```

Comment:
Unfortunately this doesn't work in the sense that it re-introduces the memory leak :( But doing a ``copy`` is really inefficient. Anyone have better ideas for dealing with this?